### PR TITLE
Feature/storage ftp service

### PR DESCRIPTION
**Pull request by mrdionjr**
I've update MountSystem to recursively create folder without a flag. What's about using the the same method for the ftp service.

### CODE REVIEW

File: src/Storage/MountFilesystem.php [hunk 1]
```diff
@@ -23,14 +30,18 @@ class MountFilesystem implements FilesystemInterface
     public function __construct($basedir)
     {
         $this->basedir = realpath($basedir);
+        $this->currentWorkingDir = $this->basedir;
```
Comment (papac) [inline src/Storage/MountFilesystem.php]: The variable name must be in snake case.

File: src/Storage/MountFilesystem.php [hunk 2]
```diff
@@ -23,14 +30,18 @@ class MountFilesystem implements FilesystemInterface
...
+
+        // Set the root folder
+        chdir($this->basedir);
```
Comment (papac) [inline src/Storage/MountFilesystem.php]: Can you remane `$this->basedir` to `$this->base_directory` ?

File: src/Storage/MountFilesystem.php [hunk 3]
```diff
@@ -163,26 +174,49 @@ public function directories($dirname)
...
+            if (false === $this->makeActualDirectory($directory, $mode)) {
+                chdir($this->basedir);
+                return false;
+            }
+            chdir($directory);
+            $this->currentWorkingDir = getcwd();
         }
 
-        $dirname = $this->path($dirname);
+        chdir($this->basedir);
```
Comment (papac) [inline src/Storage/MountFilesystem.php]: Change `$this->basedir` to `$this->base_directory`.

File: src/Storage/MountFilesystem.php [hunk 4]
```diff
@@ -163,26 +174,49 @@ public function directories($dirname)
...
+    /**
+     * Create a directory.
+     *
+     * @param $dirname
+     * @param $mode
+     * @return bool
+     */
+    private function makeActualDirectory($dirname, $mode)
+    {
+        $listing = glob($this->currentWorkingDir . '/*', GLOB_ONLYDIR) ?: [];
```
Comment (papac) [inline src/Storage/MountFilesystem.php]: Change `$this->currentWorkingDir` to `$this->current_working_directory`.

**Comment by mrdionjr** [comment]
No problem ! Will fix it in the next PR

**Pull request merged**